### PR TITLE
[4250] Update home page tiles

### PR DESCRIPTION
**Pull request by AbigailMcP**
### Context

https://trello.com/c/Ln1D3FQW/4250-m-update-home-page-tiles

### Changes proposed in this pull request

- Update Apply link to show number out of total drafts
- Update the content on the create a trainee link
- Remove link to registered trainees
- Update the home view to display the new boxes:
  - Add new Trainee scopes to use for counts
  - Update the HomeView view object to return us boxes with the statuses, counts and links
  - Update the badge component to iterate over these and map to trainee status cards
  - Update TraineeStatusCard component to decide on colour for each status and render html

Screenshot 2022-06-23 at 15 45 42

### Guidance to review

- Check out the home page and compare to the prototype
- Check that all boxes link to the correctly filtered trainee list (check the URL)

### Important business

Dependent on https://github.com/DFE-Digital/register-trainee-teachers/pull/2441 and https://github.com/DFE-Digital/register-trainee-teachers/pull/2450

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

File: app/models/trainee.rb [hunk 1]
```diff
@@ -194,6 +194,14 @@ class Trainee < ApplicationRecord
     )
   end)
 
+  scope :course_not_yet_started, -> { not_draft.where("itt_start_date > ?", Time.zone.now) }
```
Comment (AbigailMcP) [inline app/models/trainee.rb:197]: `course_not_yet_started ` and `in_training ` lifted from this PR: https://github.com/DFE-Digital/register-trainee-teachers/pull/2450 which includes tests.

**Comment by AbigailMcP** [comment]
[[M] Update home page tiles](https://trello.com/c/Ln1D3FQW/4250-m-update-home-page-tiles)

File: app/components/badges/view.html.erb [hunk 2]
```diff
@@ -1,11 +1,10 @@
...
-                                                   count: count,
-                                                   target: trainees_path(state: map_state_to_filter_params(state))) %>
-        <% end %>
-    </div>
+<div class="govuk-grid-column-full">
+  <div class="app-home-statuses">
+      <% badges.map do |badge| %>
+          <%= render TraineeStatusCard::View.new(status: badge.status,
+                                                 count: badge.trainee_count,
+                                                 target:badge.link
```
Comment (HettieS) [inline app/components/badges/view.html.erb:6]:
```suggestion
                                                 target: badge.link
```

File: app/view_objects/home_view.rb [hunk 3]
```diff
@@ -1,23 +1,19 @@
...
-    @registered_state_counts ||= state_counts.except("draft")
+  def initialize(trainees)
+    @trainees = Trainees::Filter.call(trainees: trainees, filters: {})
+    create_badges
   end
 
-  def registered_trainees_count
-    @registered_trainees_count ||= registered_state_counts.values.sum
-  end
+  Badge = Struct.new(:status, :trainee_count, :link)
```
Comment (HettieS) [inline app/view_objects/home_view.rb:13]: nice use of a struct for this :)

File: app/components/trainee_status_card/view.rb [hunk 4]
```diff
@@ -2,34 +2,28 @@
 
 module TraineeStatusCard
   class View < GovukComponent::Base
-    attr_reader :state, :count, :target
+    attr_reader :status, :count, :target
 
     STATUS_COLOURS = {
-      draft: "grey",
```
Comment (HettieS) [inline app/components/trainee_status_card/view.rb:8]: are we removing all these different statuses completely? My understanding is this is the tag on the trainee card itself (might have got that wrong)

I guess i'm wondering if we definitely don't want to show some of these labels anymore i.e. withdrawn

File: app/view_objects/home_view.rb [hunk 5]
```diff
@@ -28,35 +24,52 @@ def draft_apply_trainees_count
...
+      ),
+
+      Badge.new(
+        :deferred,
+        trainees.deferred.count,
+        trainees_path(status: %w[deferred]),
+      ),
 
-    @state_counts = counts
+      Badge.new(
```
Comment (HettieS) [inline app/view_objects/home_view.rb:50]: I'm wondering, will this incomplete filter only include registered trainees? Trying to have a look in the code although it's not super clear to me

File: spec/view_objects/home_view_spec.rb [hunk 6]
```diff
@@ -3,121 +3,73 @@
...
+              status: %w[awarded],
+              end_year: "#{current_academic_cycle.start_year} to #{current_academic_cycle.end_year}",
+            ),
+          },
+          {
+            status: :deferred,
+            trainee_count: 2,
+            link: trainees_path(status: %w[deferred]),
+          },
+          # i.e. the not_started_trainee who is still in draft
```
Comment (HettieS) [inline spec/view_objects/home_view_spec.rb:62]: should a draft trainee be showing up in these filters? Feels like they shouldn't as these tiles are for registered trainees

Comment (AbigailMcP) [reply spec/view_objects/home_view_spec.rb:62]: Ah i guess this is the same point as above! I didn't think about it, but the incomplete count should actually be "incomplete registered" (just checked the prototype too), will update.

File: app/components/trainee_status_card/view.rb (hunk 4 above)
Comment (AbigailMcP) [reply app/components/trainee_status_card/view.rb:8]: It's actually the homepage square. If you go to /view_components and find the component you'll see:

Screenshot 2022-06-27 at 11 04 16

You might be thinking of `StatusTag` which is:

Screenshot 2022-06-27 at 11 06 08

These are unchanged, we're still showing them all.

Comment (HettieS) [reply app/components/trainee_status_card/view.rb:8]: Ah amazing! Sorry my mistake :)

File: config/locales/en.yml [hunk 7]
```diff
@@ -889,10 +895,9 @@ en:
       heading: Your trainee teachers
       draft_heading: Draft trainees
       draft_trainees_link: View all draft trainees (%{count})
-      draft_apply_trainees_link: View draft trainees imported from Apply (%{count})
-      new_trainee_link: Create a new trainee record
+      draft_apply_trainees_link: View draft trainees imported from Apply (%{count} out of %{total} draft trainees)
```
Comment (edwardhorsford) [inline config/locales/en.yml:898]: In the prototype, we only show this extra bit where the apply count is different than the draft count.

**Review by edwardhorsford — APPROVED** [review]

File: config/locales/en.yml (hunk 7 above)
Comment (AbigailMcP) [reply config/locales/en.yml:898]: Should be fixed now 👍 in this commit https://github.com/DFE-Digital/register-trainee-teachers/pull/2460/commits/893fbf2e27c1600a81f60fe45f89ec67aa4e3a23

**Pull request merged**